opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -2949,12 +2955,34 @@ namespace Opm {
         }
 
         const auto qs = WellGroupHelpers::
-            getRateVector(this->well_state_, this->phase_usage_, wname);
+            getWellRateVector(this->well_state_, this->phase_usage_, wname);
 
         this->getGuideRateValues(qs, well.isInjector(), wname, grval);
 
         return grval;
     }
+    template <typename TypeTag>
+    data::GuideRateValue
+    BlackoilWellModel<TypeTag>::
+    getGuideRateInjectionGroupValues(const Group& group) const
+    {
+        auto grval = data::GuideRateValue{};
+
+        assert (this->guideRate_ != nullptr);
```

Comment:
According to Joakim's principle on assert: https://github.com/OPM/opm-simulators/issues/3117 - which is still undisputed, this should be a `std::logic_error()`

Reply:
I guess it is mainly for convenience. To avoid issues in parallel we currently need to pass around the `deferred_logger` which could be a pain sometimes.  I can give it a try for cases in this PR.

Reply:
OK; but the problem is only relevant when/if we want to log the exception? If we let the exception bubble all the way up and kill the process there should be no difference (i.e. entire application terminates) both for `assert()` and `throw`?